client/dashboard/app/queries/site-domains.ts

```diff
@@ -1,8 +1,20 @@
-import { queryOptions } from '@tanstack/react-query';
-import { fetchSiteDomains } from '../../data/site-domains';
+import { queryOptions, mutationOptions } from '@tanstack/react-query';
+import { fetchSiteDomains, setPrimaryDomain } from '../../data/site-domains';
+import { queryClient } from '../query-client';
+import { siteQueryFilter } from './site';
 
 export const siteDomainsQuery = ( siteId: number ) =>
 	queryOptions( {
 		queryKey: [ 'site', siteId, 'domains' ],
 		queryFn: () => fetchSiteDomains( siteId ),
 	} );
+
+export const createSetPrimaryDomainMutation = () =>
```

Comment:
Why is mutation name a bit weird with `create` here 🤔 why not simply `sitePrimaryDomainMutation()`

Reply:
It was originally named `setPrimaryDomainMutation`, and I wanted to use the same name at the component level, so I temporarily renamed it with a `create` prefix.

I'll go ahead and rename it to `siteSetPrimaryDomainMutation` to avoid the naming conflict. That said, I think we'll likely run into similar issues again in the future, and using a `create` prefix still seems like a reasonable convention to me 🤔